tests/openai/test_openai_init.py

```diff
@@ -0,0 +1,17 @@
+import os
+
+from mlflow.openai import _OAITokenHolder
+
+
+def test_set_api_key_on_tokenholder_init(monkeypatch):
+    # if the user sets the API key after the openai module,
+    # expect `openai.api_key` to not be set.
+    assert "OPENAI_API_KEY" not in os.environ
```

Comment:
```suggestion
    monkeypatch.delenv("OPENAI_API_KEY")
    assert "OPENAI_API_KEY" not in os.environ
```

Can we call `monkeypatch.delenv` in case we run this when `OPENAI_API_KEY` is set.

Reply:
```suggestion
    monkeypatch.delenv("OPENAI_API_KEY", raising=False)
    assert "OPENAI_API_KEY" not in os.environ
```

in case `OPENAI_API_KEY` doesn't exist.